transcript/src/main/java/com/transcriptanalyzer/transcript/Tree.java

```diff
@@ -71,17 +78,95 @@ public Node hasChild(String intent){
             }
             return null;
         }
+
+        // return a hashmap mapping the 3 leaf intents with most occurrences to their occurrences in descending order
+        public LinkedHashMap<String, Integer> getBestLeafData() {
```

Comment:
The documentation is fantastic here! It was super easy to follow along with what the function is supposed to be doing at each step - definitely will be helpful if there is any debugging to do in the future!